packages/eslint-plugin-gestalt/src/no-box-useless-props.js

```diff
@@ -8,16 +8,73 @@ import { type ESLintRule } from './eslintFlowTypes.js';
 export const errorMessages = {
   fit: '`fit` sets `maxWidth`, so `maxWidth` should not be specified when `fit` is used',
   flex:
-    '`direction`, `smDirection`, `mdDirection`, `lgDirection`, and `wrap` must be used with `display="flex"`',
-  flexGrid:
-    '`alignContent`, `alignItems`, and `justifyContent` must be used with `display="flex"`. You can suppress this error if dangerously setting `display="grid"`',
+    '`direction`, `smDirection`, `mdDirection`, `lgDirection`, and `wrap` must be used with `display="flex", or dangerously set display of "inline-flex", "inline-grid", or "grid"`',
 };
 
 const displayPropNames = ['display', 'smDisplay', 'mdDisplay', 'lgDisplay'];
 
-// These are valid for both flexbox and grid layouts
-const flexGridPropNames = ['alignContent', 'alignItems', 'justifyContent'];
-const flexPropNames = ['direction', `smDirection`, `mdDirection`, `lgDirection`, 'wrap'];
+const flexPropNames = [
+  'alignContent',
+  'alignItems',
+  'direction',
+  'justifyContent',
+  'wrap',
+  `lgDirection`,
+  `mdDirection`,
+  `smDirection`,
+];
+const dangerousFlexGridDisplays = ['inline-flex', 'grid', 'inline-grid'];
+
+function getAttributeName(attributeName): ?string {
+  return attributeName?.name;
+}
+
+function getExpressionValues(valueExpression): $ReadOnlyArray<string> {
+  return [valueExpression.consequent, valueExpression.alternate].map((option) => option.value);
+}
+
+function getAttributeValue(attributeValue): ?(string | $ReadOnlyArray<string>) {
+  const staticValue = attributeValue?.value;
+  const isBooleanShorthand = attributeValue === null;
+  if (staticValue || isBooleanShorthand) {
+    return staticValue;
+  }
+
+  const valueExpression = attributeValue.expression;
+  if (valueExpression.type === 'ConditionalExpression') {
+    return getExpressionValues(valueExpression);
+  }
+  return undefined;
+}
+
+// $FlowExpectedError[unclear-type]
+function getDangerouslySetStyles(attributeValue): null | { [string]: Object } {
```

Comment:
@dangerismycat (nit) could we replace `Object` with a more specific type?

Reply:
We could, but that goes down a bit of a rabbit hole. It'd be great if we could find exported Flow types for eslint (I can find Typescript), but I'm not sure if it's worth our time to manually convert all the TS types to Flow.

Reply:
@dangerismycat understood - not a big priority, I was more wondering if we looked into it.